packages/tse-backend/src/modules/trading-strategy/strategies/market-making/market-making.strategy.ts

```diff
@@ -0,0 +1,264 @@
+export class MarketMakingStrategy implements Strategy {
+  private logger = new Logger(MarketMakingStrategy.name);
+  private strategies: Map<number, StrategyConfig> = new Map();
+
+  constructor(
+    private readonly exchangeRegistryService: ExchangeRegistryService,
+    private readonly tradeService: ExchangeTradeService,
+    private readonly marketMakingService: MarketMakingService,
+  ) {}
+
+  async create(command: MarketMakingStrategyCommand): Promise<void> {
+    await this.marketMakingService.createStrategy({
+      userId: command.userId,
+      clientId: command.clientId,
+      pair: command.pair,
+      exchangeName: command.exchangeName,
+      bidSpread: command.bidSpread,
+      askSpread: command.askSpread,
+      orderAmount: command.orderAmount,
+      checkIntervalSeconds: command.checkIntervalSeconds,
+      numberOfLayers: command.numberOfLayers,
+      priceSourceType: command.priceSourceType,
+      amountChangePerLayer: command.amountChangePerLayer,
+      amountChangeType: command.amountChangeType,
+      ceilingPrice: command.ceilingPrice,
+      floorPrice: command.floorPrice,
+      status: StrategyInstanceStatus.RUNNING,
+    });
+  }
+
+  async pause(command: MarketMakingStrategyActionCommand): Promise<void> {
+    const strategyEntity: MarketMakingStrategyData =
+      await this.marketMakingService.findLatestStrategyByUserId(command.userId);
+    if (!strategyEntity) {
+      throw new BadRequestException('MarketMaking strategy not found');
+    }
+
+    await this.marketMakingService.updateStrategyStatusById(
+      strategyEntity.id,
+      StrategyInstanceStatus.PAUSED,
+    );
+    const strategy = this.strategies.get(strategyEntity.id);
+    if (strategy) {
+      clearInterval(strategy.intervalId);
+    }
+
+    this.logger.debug('Paused market making strategy');
+  }
+
+  async stop(command: MarketMakingStrategyActionCommand): Promise<void> {
+    const strategyEntity: MarketMakingStrategyData =
+      await this.marketMakingService.findLatestStrategyByUserId(command.userId);
+    if (!strategyEntity) {
+      throw new BadRequestException('MarketMaking strategy not found');
+    }
+
+    await this.marketMakingService.updateStrategyStatusById(
+      strategyEntity.id,
+      StrategyInstanceStatus.STOPPED,
+    );
+
+    const strategy = this.strategies.get(strategyEntity.id);
+    if (strategy) {
+      clearInterval(strategy.intervalId);
+    }
+
+    await this.cancelActiveOrders();
+
+    this.logger.debug(
+      'Stopped market making strategy, not filled orders have been canceled',
+    );
+  }
+
+  async delete(command: MarketMakingStrategyActionCommand) {
+    const strategyEntity: MarketMakingStrategyData =
+      await this.marketMakingService.findLatestStrategyByUserId(command.userId);
+    if (!strategyEntity) {
+      throw new BadRequestException('MarketMaking strategy not found');
+    }
+    if (
+      strategyEntity &&
+      strategyEntity.status !== StrategyInstanceStatus.STOPPED
```

Comment:
Why can't you delete a stopped strategy?

Reply:
Fixed: ca125584